src/sqllogictest/src/parser.rs

```diff
@@ -94,28 +94,22 @@ pub fn parse_record<'a>(
 
         // we'll follow the postgresql version of all these tests
         "skipif" => {
-            match words.next().unwrap() {
-                "postgresql" => Ok(None),
-                _ => {
-                    // query starts on the next line
-                    parse_record(mode, input)
-                }
+            if words.next().unwrap() == "postgresql" {
+                let _ = parse_record(mode, input);
             }
+            parse_record(mode, input)
         }
         "onlyif" => {
-            match words.next().unwrap() {
-                "postgresql" => {
-                    // query starts on the next line
-                    parse_record(mode, input)
-                }
-                _ => Ok(None),
+            if words.next().unwrap() != "postgresql" {
+                let _ = parse_record(mode, input);
             }
+            parse_record(mode, input)
         }
 
-        "halt" => Ok(Some(Record::Halt)),
+        "halt" => Ok(None),
```

Comment:
This looks like you just skip halts now?

Reply:
Oh, no, it's the opposite, we halt on "substest" etc below which we used to just ignore.